File: binary_search/Binary_Search.cpp
#include<iostream>

using namespace std;

int CheckMatch(int a){
	if(a!=-1){
		cout<<"Match is found";
	}else{
		cout<<"Match is not found";
	}
}


int binarySearch(int arr[],int l,int r,int x){
	if(r>=l){
		int mid = l + (r-1)/2;
		if(arr[mid]==x)
			return mid;
		if(arr[mid]>x)
			return binarySearch(arr,l,mid-1,x);	
		else
			return binarySearch(arr,mid+1,r,x);
	}
	return -1;
	//we return -1 if the element is not found
}


int main(){
	//we can either input a array by user input or 
	//for the simplicity I use the pre defined array for the search
	int ans,num;
	int arr[]={45,21,78,11,6,90};
	
	cout<<"Input the number to search : ";
	cin>>num;
	
	ans = binarySearch(arr,0,6,num);
	CheckMatch(ans);
	
	
	return 0;
}

File: binary_search/BinarySearch.py
def binarySearch (sarr, start, end, x):

    if end >= start:
       mid = start + (end- start)//2
   
       if arr[mid] == x:
          return mid
       elif arr[mid] > x:
          return binarySearchAppr(arr, start, mid-1, x)
       else:
          return binarySearchAppr(arr, mid+1, end, x)
    else:
        return -1
arr = []
sarr = []
n = int(input("No of Elements:"))
for i in range(n):
    val = int(input("Enter element %d:" %(i+1)))
    arr.append(val)          
sarr = sorted(arr)
x=int(input("Enter Search Number:"))
print ("Sorted array") 
for i in range(n): 
    print("%d" %sarr[i])
result = binarySearch(sarr, 0, len(sarr)-1, x)
if result != -1:
   print ("Element is present at index "+str(result))
else:
    print ("Element is not present in array")
